captain-definition

```diff
@@ -5,10 +5,14 @@
                 "COPY ./ /app",
                 "WORKDIR /app",
                 "RUN apt-get update",
-                "RUN apt-get install -y build-essential libpq-dev",
+                "RUN apt-get install -y build-essential libpq-dev postgresql-client cron nano gettext librsync-dev gnupg-agent",
                 "RUN pip3 install --upgrade pip",
                 "RUN pip3 install -r requirements.txt",
-                "CMD [ \"python3\", \"app.py\" ]",
+                "RUN pip3 install python-gettext duplicity b2sdk",
```

Comment:
move these to the requirements.txt file

Reply:
geting errors when moving them to requirements.txt:
```
Collecting duplicity
⠴ Building your source code...
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w3533mfb/duplicity_0bf52364136e4678ad29185382fdcee5/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w3533mfb/duplicity_0bf52364136e4678ad29185382fdcee5/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-341h4beq
         cwd: /tmp/pip-install-w3533mfb/duplicity_0bf52364136e4678ad29185382fdcee5/
    Complete output (9 lines):
    sh: 1: msgfmt: not found
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w3533mfb/duplicity_0bf52364136e4678ad29185382fdcee5/setup.py", line 281, in <module>
        data_files=get_data_files(),
      File "/tmp/pip-install-w3533mfb/duplicity_0bf52364136e4678ad29185382fdcee5/setup.py", line 123, in get_data_files
        assert not os.system(u"msgfmt po/%s.po -o po/%s/duplicity.mo" % (lang, lang)), lang
    AssertionError: bg
    Unable to get SCM version: defaulting to 0.8.19
    ----------------------------------------
```

Reply:
is there anyway to remove/ignore teh error message from pylint for the duplicity? more specifically when its placed in the requirements.txt?

Reply:
found two questions on fixing this issue from docker:
- https://dockerquestions.com/2021/03/22/dockerfile-pip-install-fails-with-requirements-txt-but-succeeds-with-individual-packages/
- https://dockerquestions.com/2020/01/07/running-pip-install-requirements-txt-fails-in-docker-build/

no answer it seems.

Reply:
so the github environment needs to install the same modules to make sure it no longer errors. added for apt-get install before installing modules using pip.